### ISSUE TITLE
Complete Add Cache Functionality

### ISSUE THREAD

**Issue by fconlon** [opensocsysarch/CoreGenPortal#82]
- [x] Criteria 1

### Criteria 1
**Given:**
* The Add Cache Window is open
* The user has entered valid data into all fields

**When:**
* The user clicks save

**Then:**
* The new Cache will be added to the project
* The project will be saved to the IR .yaml file

**Comment by fconlon** [comment]
@jleidel now that I've completed all the edit window functionality I can do all of these adds really fast with the insert/delete functions in the backend (need the delete in the even the user hits save with invalid info), but the delete functions are private. Would it be fine to elevate them to public?

**Comment by jleidel** [comment]
> @jleidel now that I've completed all the edit window functionality I can do all of these adds really fast with the insert/delete functions in the backend (need the delete in the even the user hits save with invalid info), but the delete functions are private. Would it be fine to elevate them to public?

You should use the `DeleteNode(CoreGenNode *N)` function which is exposed to publicly.  This function takes a pointer to a base `CoreGenNode` object, decodes it for its respective type and deletes it.  The process of deletion also includes cleaning up any necessary dangling links in adjacent nodes.  

So, for example: 
`CoreGenCache *Cache = CG->InsertCache("NAME", 2, 8 );`
`CG->DeleteNode(static_cast<CoreGenNode *>(Cache));`

**Comment by fconlon** [comment]
This actually Makes things even easier. I'm not sure how I missed this function last night. I did prototype it out this morning before I left for work and, with everything already set up in the edit windows, I can add each piece of add functionality with 1 line of code. This delete also means I can just have one delete at the end instead of a different one for each type of node. I'm glad I asked before I ran off into the deep blue yonder. My merge request was approved late yesterday and I'm back in the office this morning so I won't be able to work on this until I get home tonight. It shouldn't take took long to add and test each add node and then I'll focus on getting the Instruction Format drawn and work on plugins. Is there a sample yaml with a plugin already in it?
